network-api/networkapi/wagtailpages/pagemodels/buyersguide/call_to_action.py

```diff
@@ -78,16 +83,16 @@ def clean(self):
         errors = {}
         duplicate_link_target_error = ErrorList(['Please select a page OR enter a URL for the link (choose one)'])
         # If user enters both link URL and link page:
-        if self.link_label and self.link_target_url and self.link_target_page:
+        if self.link_label and (self.link_target_url and self.link_target_page):
             errors['link_target_url'] = duplicate_link_target_error
             errors['link_target_page'] = duplicate_link_target_error
-        # If user enters link URL or page but no label:
-        elif self.link_target_page or self.link_target_url and not self.link_label:
-            errors['link_label'] = ErrorList(['Please enter a label for the link'])
         # If user enters link label but no page or URL to link to:
-        elif not self.link_target_page and not self.link_target_url and self.link_label:
+        elif (not self.link_target_page and not self.link_target_url) and self.link_label:
```

Comment:
What do you think about using the same order for the condition as in the comment above? 

```suggestion
        elif self.link_label and (not self.link_target_page and not self.link_target_url):
```

Reply:
Updated!